hr_attendance_calendar/models/hr_forced_due_hours.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Samuel Fringeli <samuel.fringeli@me.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, fields, api
+
+
+class HrForcedDueHours(models.Model):
+    _name = 'hr.forced.due.hours'
+    _order = 'date'
+
+    _sql_constraints = [('unique_due_hours', 'unique(date, employee_id)',
+                         'This "Forced due hour" already exists')]
+
+    employee_id = fields.Many2one('hr.employee',
+                                  string='Employee', required=True)
+    date = fields.Date('Date', required=True)
+    forced_due_hours = fields.Float('Forced due hours', required=True)
+
+    @api.model
+    def recompute_due_hours(self, employee_id, date):
+        return self.env['hr.attendance.day'].search(
+            [('employee_id', '=', employee_id),
+             ('date', '=', date)]).recompute_due_hours()
+
+    @api.one
+    def write(self, vals):
+        old_vals = dict()
+        for field in ['employee_id', 'date', 'forced_due_hours']:
+            old_vals[field] = getattr(self, field)
+
+        res = super(HrForcedDueHours, self).write(vals)
+
+        self.recompute_due_hours(
+            old_vals['employee_id'].id, old_vals['date'])
```

Comment:
old_vals['employee_id'] should already contain an id and not a record, so remove `.id`

Reply:
No, because `old_vals['employee_id'] = self.employee_id` that is an `hr.employee()` object

Reply:
If you do that it's wrong! The write method never takes records as parameters in the dictionary. It should always be ids.